sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -823,44 +750,80 @@ public String getEntityPath() {
     }
 
     /**
-     * Asynchronously renews the lock on the specified message. The lock will be renewed based on the setting specified
+     * Asynchronously renews the specified message. The lock will be renewed based on the setting specified
      * on the entity. When a message is received in {@link ReceiveMode#PEEK_LOCK} mode, the message is locked on the
-     * server for this receiver instance for a duration as specified during the Queue creation (LockDuration). If
+     * server for this receiver instance for a duration as specified during the entity creation (LockDuration). If
      * processing of the message requires longer than this duration, the lock needs to be renewed. For each renewal, the
      * lock is reset to the entity's LockDuration value.
      *
-     * @param lockToken Lock token of the message to renew.
+     * @param message The {@link ServiceBusReceivedMessage} to perform auto-lock renewal.
      *
      * @return The new expiration time for the message.
-     * @throws NullPointerException if {@code lockToken} is null.
+     * @throws NullPointerException if {@code message} is null.
      * @throws UnsupportedOperationException if the receiver was opened in {@link ReceiveMode#RECEIVE_AND_DELETE}
      *     mode.
      * @throws IllegalStateException if the receiver is a session receiver.
-     * @throws IllegalArgumentException if {@code lockToken} is an empty value.
+     * @throws IllegalArgumentException if {@code message.getLockToken()} is an empty value.
      */
-    public Mono<OffsetDateTime> renewMessageLock(String lockToken) {
+    public Mono<OffsetDateTime> renewMessageLock(ServiceBusReceivedMessage message) {
         if (isDisposed.get()) {
             return monoError(logger, new IllegalStateException(
                 String.format(INVALID_OPERATION_DISPOSED_RECEIVER, "renewMessageLock")));
-        } else if (Objects.isNull(lockToken)) {
-            return monoError(logger, new NullPointerException("'lockToken' cannot be null."));
-        } else if (lockToken.isEmpty()) {
-            return monoError(logger, new IllegalArgumentException("'lockToken' cannot be empty."));
+        } else if (Objects.isNull(message)) {
+            return monoError(logger, new NullPointerException("'message' cannot be null."));
+        } else if (message.getLockToken().isEmpty()) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("'message.getLockToken()' cannot be empty."));
```

Comment:
The above uses monoError. Why is this line different?

Reply:
Good catch. I've replaced all instances